core/tests/unit/ldml/ldml.cpp

```diff
@@ -118,11 +122,23 @@ apply_action(
           text_store.pop_back();
         }
       }
-      assert(ch == act.backspace.expected_value);
-
-      assert(context.back().type == KM_CORE_CT_CHAR);
-      assert(context.back().character == ch);
-      context.pop_back();
+      if (act.backspace.expected_type == KM_CORE_BT_CHAR) {
+        assert(ch == act.backspace.expected_value);
+        assert(context.back().type == KM_CORE_CT_CHAR);
+        assert(context.back().character == ch);
+        context.pop_back();
+      } else {
+        assert(act.backspace.expected_type == KM_CORE_BT_UNKNOWN); // else it must be unknown
```

Comment:
BT_UNKNOWN should only be used when the context is empty.

Reply:
added an assert for that request